source/write-operations/bulk-writes.txt

```diff
@@ -1,58 +1,347 @@
-.. _javars-bulk-writes:
+.. _java-rs-bulk-write:
 
 =====================
 Bulk Write Operations
 =====================
 
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 2
+   :class: singlecol
+
 .. facet::
    :name: genre
    :values: reference
 
 .. meta::
-   :keywords: code example, multiple changes
-
-Starting in v2.6, MongoDB supports bulk write commands
-for insert, update, and delete operations in a way that allows the
-driver to implement the correct semantics for ``BulkWriteResult`` and
-``BulkWriteException``.
-
-There are two types of bulk operations, ordered and unordered bulk
-operations:
-
-1. Ordered bulk operations execute all the operations in order and
-   error out on the first write error.
-#. Unordered bulk operations execute all the operations and report any
-   the errors. Unordered bulk operations do not guarantee an order of
-   execution.
-
-.. include:: /includes/subscriber-note.rst
-
-The following code provides examples using ordered and unordered
-operations:
-
-.. code-block:: java
-
-   // Ordered bulk operation - order is guaranteed
-   collection.bulkWrite(
-       Arrays.asList(new InsertOneModel<>(new Document("_id", 4)),
-           new InsertOneModel<>(new Document("_id", 5)),
-           new InsertOneModel<>(new Document("_id", 6)),
-           new UpdateOneModel<>(new Document("_id", 1),
-                                new Document("$set", new Document("x", 2))),
-           new DeleteOneModel<>(new Document("_id", 2)),
-           new ReplaceOneModel<>(new Document("_id", 3),
-                                 new Document("_id", 3).append("x", 4))))
-   .subscribe(new ObservableSubscriber<BulkWriteResult>());
-    
-   // Unordered bulk operation - no guarantee of order of operation
-   collection.bulkWrite(
-       Arrays.asList(new InsertOneModel<>(new Document("_id", 4)),
-           new InsertOneModel<>(new Document("_id", 5)),
-           new InsertOneModel<>(new Document("_id", 6)),
-           new UpdateOneModel<>(new Document("_id", 1),
-                                new Document("$set", new Document("x", 2))),
-           new DeleteOneModel<>(new Document("_id", 2)),
-           new ReplaceOneModel<>(new Document("_id", 3),
-                                 new Document("_id", 3).append("x", 4))),
-       new BulkWriteOptions().ordered(false))
-   .subscribe(new ObservableSubscriber<BulkWriteResult>());
+   :keywords: insert, update, replace, code example
+
+Overview
+--------
+
+Consider a scenario in which you want to insert a document into a collection,
+update multiple other documents, then delete a document. If you use
+individual methods, each operation requires its own database call. This guide
+shows you how to use bulk write operations to perform multiple write operations
+in a single database call.
+
+Sample Data
+~~~~~~~~~~~
+
+The examples in this guide use the ``sample_restaurants.restaurants`` collection
```

Comment:
S: [nit] since this is a namespace rather than a collection, reword to something like:
```suggestion
The examples in this guide use the ``restaurants`` collection in the ``sample_restaurants`` database
```

Reply:
Because the other write/read pages, such as: https://preview-mongodbshuangela.gatsbyjs.io/java-rs/DOCSP-39699-bulk-write/write-operations/insert-documents/ use sample_restaurants.restaurants and this is a convention I've established, I'm going to keep it as is